include/Operator/SecondQuantized/USQHamiltonian.hpp

```diff
@@ -0,0 +1,250 @@
+
+    std::vector<ScalarSQTwoElectronOperator<Scalar>> two_op_mixed;  // alpha & beta mixed operators represented as g_aabb
+    ScalarSQTwoElectronOperator<Scalar> total_two_op_mixed;  // the total alpha & beta mixed operator represented as g_aabb
+
+public:
+
+    /*
+     *  CONSTRUCTORS
+     */
+    USQHamiltonian() = default;
+
+    /**
+     *  @param sq_hamiltonian_alpha      alpha hamiltonian
+     *  @param sq_hamiltonian_beta       beta hamiltonian
+     *  @param two_op_mixed              alpha & beta mixed operators
+     */
+    USQHamiltonian(const SQHamiltonian<Scalar>& sq_hamiltonian_alpha, const SQHamiltonian<Scalar>& sq_hamiltonian_beta, const std::vector<ScalarSQTwoElectronOperator<Scalar>>& two_op_mixed) :
+        sq_hamiltonian_alpha (sq_hamiltonian_alpha),
+        sq_hamiltonian_beta (sq_hamiltonian_beta),
+        two_op_mixed (two_op_mixed)
+    {
+
+        const auto dim = sq_hamiltonian_alpha.dimension();
+
+        if (sq_hamiltonian_beta.dimension() != dim) {
+            throw std::invalid_argument("USQHamiltonian::USQHamiltonian(const SQHamiltonian<Scalar>& sq_hamiltonian_alpha, const SQHamiltonian<Scalar>& sq_hamiltonian_beta, const ScalarSQTwoElectronOperator<Scalar>& two_op_mixed): The dimensions of the alpha and beta Hamiltonian are incompatible");
+        }
+        
+        for (const auto& two_op : this->two_op_mixed) {
+            if (two_op.dimension() != dim) {
+                throw std::invalid_argument("USQHamiltonian::USQHamiltonian(const SQHamiltonian<Scalar>& sq_hamiltonian_alpha, const SQHamiltonian<Scalar>& sq_hamiltonian_beta, const ScalarSQTwoElectronOperator<Scalar>& two_op_mixed): The dimensions of the mixed two electron operator are incompatible with the Hamiltonian");
+            }
+        }
+        
+        // Calculate the total two-electron operator
+        QCRankFourTensor<Scalar> total_two_op_par (dim);
+        total_two_op_par.setZero();
+        for (const auto& two_op : this->two_op_mixed) {
+            total_two_op_par += two_op.parameters().Eigen();
+        }
+        this->total_two_op_mixed = ScalarSQTwoElectronOperator<Scalar>({total_two_op_par});
+    }
+
+    /**
+     *  @param sq_hamiltonian_alpha      alpha hamiltonian
+     *  @param sq_hamiltonian_beta       beta hamiltonian
+     *  @param two_op_mixed              alpha & beta mixed operator
+     */
+    USQHamiltonian(const SQHamiltonian<Scalar>& sq_hamiltonian_alpha, const SQHamiltonian<Scalar>& sq_hamiltonian_beta, const ScalarSQTwoElectronOperator<Scalar>& two_op_mixed) :
+        USQHamiltonian(sq_hamiltonian_alpha, sq_hamiltonian_beta, std::vector<ScalarSQTwoElectronOperator<Scalar>>({two_op_mixed}))
+    {}
+
+
+    /*
+     *  NAMED CONSTRUCTORS
+     */
+
+    /**
+     *  Construct the molecular Hamiltonian in a given single-particle basis
+     *
+     *  @param sp_basis     the single-particle basis in which the Hamiltonian should be expressed
```

Comment:
You're missing `sp_basis_alpha` and `sp_basis_beta`, and also `molecule`.

Eventually, this should become `sp_basis`, but whose type is `USpinorBasis`. Could you mention this in the previous issue that I've mentioned?